service-search-nls/src/main/java/fi/nls/oskari/search/channel/OgcApiSearchChannel.java

```diff
@@ -0,0 +1,202 @@
+package fi.nls.oskari.search.channel;
+
+import fi.mml.portti.service.search.ChannelSearchResult;
+import fi.mml.portti.service.search.SearchCriteria;
+import fi.mml.portti.service.search.SearchResultItem;
+import fi.nls.oskari.annotation.Oskari;
+import fi.nls.oskari.log.LogFactory;
+import fi.nls.oskari.log.Logger;
+import fi.nls.oskari.search.channel.SearchChannel;
+import fi.nls.oskari.util.IOHelper;
+import fi.nls.oskari.util.PropertyUtil;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+
+
+@Oskari(OgcApiSearchChannel.ID)
+public class OgcApiSearchChannel extends SearchChannel {
+
+
+    public static final String ID = "OGC_API_SEARCH_CHANNEL";
+
+    //Local configs
+    /**
+     * One can use property search.channel.OGC_API_SEARCH_CHANNEL.apikey but if really needed apikey can go here.
+     */
+    private static String apiKey = null;
+    private static String searchAddress = "https://avoin-paikkatieto.maanmittauslaitos.fi/geographic-names/features/v1/collections/placenames/items?spelling_case_insensitive=";
+    private static String additionalPlaceTypeInfoBaseUrl = "https://beta-paikkatieto.maanmittauslaitos.fi/catalogue/v1/codelistgroups/geographic-names/codelists/";
+    private static String[] localAdditionalInfoNames = {"placeType", "municipality"};
+    /**
+     * Set true if one wants to use local jsons instead of url based one
+     */
+    private static final boolean USE_LOCAL_ADDITIONAL_INFO = true;
+    /**
+     * Add languages you have translaited here as language code
+     */
+    private static String[] languageArray = {"fin", "eng", "swe"};
+
+    private static HashMap<Integer, HashMap<String, String>> placetypeCodes;
+    private static HashMap<Integer, HashMap<String, String>> municipalityCodes;
+
+    private Logger log = LogFactory.getLogger(this.getClass());
+
+    private static String CRS = null;
+    private static final String defaultCRS = "http://www.opengis.net/def/crs/EPSG/0/3067";
+
+
+    public ChannelSearchResult doSearch(SearchCriteria criteria) {
+        ChannelSearchResult result = new ChannelSearchResult();
+
+        if (placetypeCodes == null) {
+            placetypeCodes = buildAdditionalInfo(localAdditionalInfoNames[0]);
+        }
+        if (municipalityCodes == null) {
+            municipalityCodes = buildAdditionalInfo(localAdditionalInfoNames[1]);
+        }
+
+        //ApiKey Things
+        String propertyKey = PropertyUtil.get("search.channel." + ID + ".apikey");
```

Comment:
There getProperty() in the SearchChannel baseclass that makes this easier:
```suggestion
        String propertyKey = getProperty("apikey");
```
Also see: https://github.com/nlsfi/oskari-server-extras/blob/develop/service-search-nls/src/main/java/fi/nls/oskari/search/NLSFIGeocodingSearchChannel.java#L62 which uses a `APIkey` instead of `apikey` for the same purpose and also allows configuring apikey as `service.user` (user is used in other channels that use credentials). Change this to match that.